internal/webhooks/metal3.io/v1alpha1/baremetalhost_validation.go

```diff
@@ -109,8 +109,26 @@ func (webhook *BareMetalHost) validateChanges(oldObj *metal3api.BareMetalHost, n
 		errs = append(errs, errors.New("BMC address can not be changed if the BMH is not in the Registering state, or if the BMH is not detached"))
 	}
 
-	if oldObj.Spec.BootMACAddress != "" && newObj.Spec.BootMACAddress != oldObj.Spec.BootMACAddress {
-		errs = append(errs, errors.New("bootMACAddress can not be changed once it is set"))
+	if oldObj.Spec.BootMACAddress != "" {
+		// BootMACAddress is immutable once set - cannot be changed or cleared
+		if newObj.Spec.BootMACAddress == "" {
+			errs = append(errs, errors.New("bootMACAddress can not be unset once it is set"))
```

Comment:
I'm not so sure how necessary it is to handle this special case. We would catch it anyway when validating it below. But I guess this gives a nicer message to the users, so from that perspective it is good.